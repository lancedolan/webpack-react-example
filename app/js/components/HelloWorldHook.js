import React, { useState, useEffect } from 'react';
import css from '../../css/hello-world.scss'

export default function HelloWorldHook(props) {

    useEffect(()=>{
        // do things on render

        return function cleanup() {
            //do cleanup when done rendering (perhaps unmounting, perhaps re-rendering)
        }
    }, [])

    return <div className="hello">Hello world from hook?</div>;
}