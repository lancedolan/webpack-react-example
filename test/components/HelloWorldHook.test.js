import React from "react";
import render from "react-test-renderer";
import HelloWorldHook from "../../app/js/components/HelloWorldHook";

describe('ExampleHook',()=>{
    const comp = render.create(<HelloWorldHook />)
    const compJson = comp.toJSON()

    it('should have children', ()=>{
        expect(compJson.children.length).toBeGreaterThan(0)
    })

    it('should match snapshot', ()=>{
        expect(compJson).toMatchSnapshot()
    })

})
