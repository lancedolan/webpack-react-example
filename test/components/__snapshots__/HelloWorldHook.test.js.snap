// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleHook should match snapshot 1`] = `
<div
  className="hello"
>
  Hello world from hook?
</div>
`;
